dataactbroker/handlers/fileHandler.py

```diff
@@ -507,6 +552,9 @@ def checkGeneration(self, submission_id = None, file_type = None):
         """
         if submission_id is None or file_type is None:
             submission_id, file_type = self.getRequestParamsForGenerate()
+        # Check permission to submission
+        self.checkSubmissionById(submission_id, file_type)
```

Comment:
Is the logic in this `checkSubmissionById` function a good contender to user for decorating some of these functions? Or is there another way we can elevate permissions checking so that it's done before we get here?

I actually don't know that right answer..pinging @cmc333333 and @nmonga91 to see if they have .02.